src/Symfony/Component/Uid/Command/GenerateUuidCommand.php

```diff
@@ -0,0 +1,194 @@
+
+To generate a name based UUID:
+
+    <info>php %command.full_name% --name-based=foo</info>
+
+To specify a name based UUID's namespace:
+
+    <info>php %command.full_name% --name-based=bar --namespace=fb3502dc-137e-4849-8886-ac90d07f64a7</info>
+
+To generate a random based UUID:
+
+    <info>php %command.full_name% --random-based</info>
+
+To generate several UUIDs:
+
+    <info>php %command.full_name% --count=10</info>
+
+To output a specific format:
+
+    <info>php %command.full_name% --format=base_58</info>
+EOF
+            )
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output instanceof ConsoleOutputInterface ? $output->getErrorOutput() : $output);
+
+        $time = $input->getOption('time-based');
+        $node = $input->getOption('node');
+        $name = $input->getOption('name-based');
+        $namespace = $input->getOption('namespace');
+        $random = $input->getOption('random-based');
+
+        switch (true) {
+            case !$time && !$node && !$name && !$namespace && !$random:
+                $create = [$this->factory, 'create'];
+
+                break;
+            case $time && !$name && !$namespace && !$random:
+                if ($node) {
+                    try {
+                        $node = Uuid::fromString($node);
+                    } catch (\InvalidArgumentException $e) {
+                        $io->error(sprintf('Invalid node "%s". It cannot be parsed.', $node));
+
+                        return 1;
+                    }
+                }
+
+                try {
+                    $time = new \DateTimeImmutable($time);
+                } catch (\Exception $e) {
+                    $io->error(sprintf('Invalid timestamp "%s". It cannot be parsed.', $input->getOption('time-based')));
+
+                    return 2;
+                }
+
+                if ($time < new \DateTimeImmutable('@-12219292800')) {
+                    $io->error(sprintf('Invalid timestamp "%s". It must be greater than or equals to the UUID epoch (1582-10-15 00:00:00).', $input->getOption('time-based')));
+
+                    return 3;
+                }
+
+                $create = function () use ($node, $time): Uuid { return $this->factory->timeBased($node)->create($time); };
+
+                break;
+            case $name && !$time && !$node && !$random:
+                if ($namespace) {
+                    try {
+                        $namespace = Uuid::fromString($namespace);
+                    } catch (\InvalidArgumentException $e) {
+                        $io->error(sprintf('Invalid namespace "%s". It cannot be parsed.', $namespace));
+
+                        return 4;
+                    }
+                } else {
+                    $refl = new \ReflectionProperty($this->factory, 'nameBasedNamespace');
```

Comment:
not sure why we'd need this - the factory already checks that
can't we do without?

Reply:
The factory throws a \LogicException "A namespace should be defined when using UuidFactory::nameBased()". As an average user, when I use the command, I probably do not know about the factory, if I see this error message it is not going to help me a lot.